packages/main-api/tsconfig.json

```diff
@@ -0,0 +1,70 @@
+{
```

Comment:
With a monorepo we should be inheriting from the base tsconfig.json file in the project, then extending that where needed. 

https://github.com/Portaler-Zone/portaler-core/blob/main/packages/frontend/tsconfig.json

This actually just started as:

```
{
  "extends": "../../tsconfig.json",
}
```

But when I ran it for the first time I think CRA or maybe TS gave some kind of message I didn't quite catch before it went away and it added the rest of the file.

Reply:
This might mean we need to move some of the JSX or Node specific configs into the individual packages, I've never done a Node TS repo before, but that should be easy enough.

Reply:
I get "unknown compiler option" for "extends".

Reply:
Ok, I'll take a look at this after work. I actually ran into the same issue but I thought I fixed it. I forget exactly what it was, but it was something stupid.